### PR TITLE
Use isotp socket raw

### PR DESCRIPTION
**Pull request by tictacmenthe**
Added an isotp can socket adapter to the project.
Integrated it to the candump/read_bus utility, and to the main program.

TODO: add other boards' adapters, and make it easy to configure their addresses.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,7 +32,7 @@ clean:
 
 .PHONY: candump
 candump:
-	@pipenv run python -m tool.script.read_bus $(port)
+	@pipenv run python -m tool.script.read_bus $(dev) $(id_rx) $(id_tx)
```
Comment (bonnetn) [inline Makefile:35]: I'd prefer to have the read_bus being able to read via TCP, it's useful for me to debug remotely, and we don't really care about 1s freeze in debug mode do we?

File: highlevel/adapter/socket/isotp.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            self.socket.bind(self.address)
+            transport, protocol = await asyncio.get_event_loop(
+            ).create_datagram_endpoint(sock=self.socket,
+                                       protocol_factory=lambda: self)
+            # DatagramProtocol methods are typed with BaseTransport transports, but are actually
+            # DatagramTransports, that have methods we need here. So we ignore the mypy errors.
+            self.transport = transport  # type: ignore
+            self.protocol = protocol  # type: ignore
+            while True:
+                await asyncio.sleep(1)
```
Comment (bonnetn) [inline highlevel/adapter/socket/isotp.py:55]: nit: you can sleep for a much longer time

File: highlevel/adapter/socket/isotp.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        except ConnectionRefusedError:
+            raise RuntimeError("Error on datagram endpoint creation:" +
```
Comment (bonnetn) [inline highlevel/adapter/socket/isotp.py:58]: I would just raise? or not catch? 

and add logging, just so we don't "eat" the exception
image

File: highlevel/adapter/socket/tcp.py [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
+from highlevel.logger import LOGGER
 
 
 class TCPSocketAdapter(SocketAdapter):
     """
     Sends and receive messages from a TCP/IP socket, with the messages being encoded in ASCII:
-    <ABCD>
-    
-    ABCD is the the hexadecimal representation of the data surrounded by < and >.
+    <ABCD>\n
```
Comment (bonnetn) [inline highlevel/adapter/socket/tcp.py:16]: \n? :P

File: Makefile (hunk 1 above)
Comment (tictacmenthe) [reply Makefile:35]: We have to test on the pi more extensively to check that those billions of isotpservers dont generate any issue, I would rather take them out first, then verify this.

Though, even if they cause issues, we could start them for debug purposes once in a while.

File: highlevel/adapter/socket/isotp.py (hunk 2 above)
Comment (tictacmenthe) [reply highlevel/adapter/socket/isotp.py:55]: does it matter in any way?

File: highlevel/adapter/socket/tcp.py (hunk 4 above)
Comment (tictacmenthe) [reply highlevel/adapter/socket/tcp.py:16]: The TCP packets used by the isotpserver include the line break..

Comment (bonnetn) [reply highlevel/adapter/socket/tcp.py:16]: ha but we don't really take them into account, we could very well not have them

File: Makefile (hunk 1 above)
Comment (bonnetn) [reply Makefile:35]: fine, wateva

**Pull request merged**